### ISSUE TITLE
Unsatisfying result

### ISSUE THREAD

**Issue by don-tpanic** [yftah89/structural-correspondence-learning-SCL#1]
First of all, thanks for the code I find it really helpful. After running it the result I got is not as good as the result in the original paper. Would you please help to explain what might be the reason here as by going through the code it all looks perfect to me.

(adapt from `book` to `kitchen`)
dim = 50  on dev : rep = 0.7025 , non = 0.7825 all = 0.79
, on target: rep = 0.666 , non = 0.7555 all = 0.7575 c_parm = 0.1

dim = 100  on dev : rep = 0.7125 , non = 0.7825 all = 0.785
, on target: rep = 0.6925 , non = 0.7555 all = 0.7625 c_parm = 0.1

dim = 150  on dev : rep = 0.745 , non = 0.7825 all = 0.795
, on target: rep = 0.7135 , non = 0.7555 all = 0.75 c_parm = 0.1

**Comment by yftah89** [comment]
Hi Don, thanks for your question :)
Blitzer's article is extremely  strait forward -  https://www.cs.jhu.edu/~mdredze/publications/sentiment_acl07.pdf

As you said, I believe that my implementation is very true to blitzer's idea, I can't really speak on his behalf, I don't think he cheated in any way, the performance differences surprised me as well and they can steam from wide range of reasons such as : different preprocessing, different train/dev splits and so on. 

As both my articles (https://www.aclweb.org/anthology/K/K17/K17-1040.pdf; http://www.aclweb.org/anthology/N18-1112) heavily relays on Blitzer's idea  I wanted to use it as baseline so I implemented his article (with my own minimalist pre-processing and training size) for fair comparison.

**Comment by don-tpanic** [comment]
Hi Yftah,

Yes I really think your code is correct but it's all frustrating that the result seems to suggest that SCL has almost no positive impact on the accuracy (less than 1%). I am quite new to this but can I argue that SCL has its limitations because if different preprocessing or train/test splits is capable of changing the result completely, it's probably not a very widely applicable method? Also although Blitzer was not being explicit about how exactly he preprocessed the data, I think the way you did it should be at least roughly the same as he suggested here and there in the paper. So i still can't believe it would make such a big difference.

Btw, could you explain in `sentiment.py`,
why did you do `X_2_test_unlabeld = bigram_vectorizer_unlabeled.transform` but not `bigram_vectorizer.fit_transform`? I guess the only difference between these two transformation is the min_df value?

The full block:
`
    trainSent = train
    bigram_vectorizer = CountVectorizer(ngram_range=(1, 2), token_pattern=r'\b\w+\b', min_df=20, binary=True)
    X_2_train = bigram_vectorizer.fit_transform(trainSent).toarray()
    X_2_test_unlabeld = bigram_vectorizer_unlabeled.transform(trainSent).toarray()
    XforREP = np.delete(X_2_test_unlabeld, pivotsCounts, 1)  # delete second column of C`


Thanks a lot for your time

**Comment by yftah89** [comment]
I sorry I didn't understand you in the first time.
you **must** use the "transform" and not "fit_transform" otherwise you won't be compatible with the learned representation.

BTW, blitzer might use tf-idf,lematization, stop-words and many more result shifting actions, I wish he gave more details in his article but each one of the above could definitely affect the results

**Comment by don-tpanic** [comment]
Yes yes and if use fit_transform, it's going to override the vectorizer learned from other parts of the data. Thanks again. I think we can close this one for now :)
